libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -267,14 +267,59 @@ void DeclarationTypeChecker::endVisit(Mapping const& _mapping)
 		solAssert(dynamic_cast<ElementaryTypeName const*>(&_mapping.keyType()), "");
 
 	Type const* keyType = _mapping.keyType().annotation().type;
+	ASTString keyName = _mapping.keyName();
+
 	Type const* valueType = _mapping.valueType().annotation().type;
+	ASTString valueName = _mapping.valueName();
 
 	// Convert key type to memory.
 	keyType = TypeProvider::withLocationIfReference(DataLocation::Memory, keyType);
 
 	// Convert value type to storage reference.
 	valueType = TypeProvider::withLocationIfReference(DataLocation::Storage, valueType);
-	_mapping.annotation().type = TypeProvider::mapping(keyType, valueType);
+	_mapping.annotation().type = TypeProvider::mapping(keyType, keyName, valueType, valueName);
+
+	// Check if parameter names are conflicting.
+	if (!keyName.empty())
+	{
+		auto childMappingType = dynamic_cast<MappingType const*>(valueType);
+		ASTString currentValueName = valueName;
+		bool isError = false;
+		while (true) {
+			// Value type is a mapping.
+			if (childMappingType)
+			{
+				// Compare top mapping's key name with child mapping's key name.
+				ASTString childKeyName = childMappingType->keyName();
+				if (keyName == childKeyName)
+				{
+					isError = true;
+					break;
+				}
```

Comment:
This is semantically different than the previous approach (which was correct), as you're now break-ing from the loop as soon as you encounter an error instead of continuing to type check, which is ultimately why you had to remove one line form [this test](https://github.com/ethereum/solidity/pull/13384/files#diff-789fefb5e21fd40f374366380ff975e1bfdd6f2b2ed9af02a3c6982e9a449e07). All you need to do here is set the boolean indicator, and continue as per usual.

```suggestion
				if (keyName == childKeyName)
					isError = true;
```

Reply:
I will make this change but I still don't see the value since the error would still be reported once. Because in previous logic it was reported every time it was discovered and now only the last error would be reported.